streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -92,69 +95,80 @@ public void validatePartitionExists(final TopicPartition topicPartition, final S
 
     @Override
     public void register(final StateRestorer restorer) {
-        if (restorer.offsetLimit() > 0) {
-            stateRestorers.put(restorer.partition(), restorer);
+        stateRestorers.put(restorer.partition(), restorer);
+    }
+
+
+    public Collection<TopicPartition> restore() {
+        if (!initialized) {
+            initialize();
+        }
+
+        if (needsRestoring.isEmpty()) {
```

Comment:
it happens if everything has already been restored or there was nothing to restore to begin with. I'm not convinced moving it into the other `if` block makes it any clearer.